tests/spin/test_spin_hamiltonian.py

```diff
@@ -782,3 +783,826 @@ def test_fermi_hubbard_hamiltonian_matrix(shape, n_cells, t, coulomb, expected_h
+            + (0.615 + 0j) * (X(1) @ Z(2) @ X(3))
+            + (0.615 + 0j) * (Y(0) @ Z(1) @ Z(2) @ Z(3) @ Y(4))
+            + (0.615 + 0j) * (X(0) @ Z(1) @ Z(2) @ Z(3) @ X(4))
+            + (0.615 + 0j) * (Y(1) @ Z(2) @ Z(3) @ Z(4) @ Y(5))
+            + (0.615 + 0j) * (X(1) @ Z(2) @ Z(3) @ Z(4) @ X(5))
+            + (0.615 + 0j) * (Y(2) @ Z(3) @ Z(4) @ Z(5) @ Y(6))
+            + (0.615 + 0j) * (X(2) @ Z(3) @ Z(4) @ Z(5) @ X(6))
+            + (0.615 + 0j) * (Y(3) @ Z(4) @ Z(5) @ Z(6) @ Y(7))
+            + (0.615 + 0j) * (X(3) @ Z(4) @ Z(5) @ Z(6) @ X(7))
+            + (0.615 + 0j) * (Y(4) @ Z(5) @ Y(6))
+            + (0.615 + 0j) * (X(4) @ Z(5) @ X(6))
+            + (0.615 + 0j) * (Y(5) @ Z(6) @ Y(7))
+            + (0.615 + 0j) * (X(5) @ Z(6) @ X(7))
+            + (8.020000000000005 + 0j) * I(0)
+            + (-2.005 + 0j) * Z(1)
+            + (-2.005 + 0j) * Z(0)
+            + (0.585 + 0j) * (Z(0) @ Z(1))
+            + (-2.005 + 0j) * Z(3)
+            + (-2.005 + 0j) * Z(2)
+            + (0.585 + 0j) * (Z(2) @ Z(3))
+            + (-2.005 + 0j) * Z(5)
+            + (-2.005 + 0j) * Z(4)
+            + (0.585 + 0j) * (Z(4) @ Z(5))
+            + (-2.005 + 0j) * Z(7)
+            + (-2.005 + 0j) * Z(6)
+            + (0.585 + 0j) * (Z(6) @ Z(7))
+            + (0.355 + 0j) * (Z(0) @ Z(2))
+            + (0.355 + 0j) * (Z(0) @ Z(3))
+            + (0.355 + 0j) * (Z(1) @ Z(2))
+            + (0.355 + 0j) * (Z(1) @ Z(3))
+            + (0.355 + 0j) * (Z(0) @ Z(4))
+            + (0.355 + 0j) * (Z(0) @ Z(5))
+            + (0.355 + 0j) * (Z(1) @ Z(4))
+            + (0.355 + 0j) * (Z(1) @ Z(5))
+            + (0.355 + 0j) * (Z(2) @ Z(6))
+            + (0.355 + 0j) * (Z(2) @ Z(7))
+            + (0.355 + 0j) * (Z(3) @ Z(6))
+            + (0.355 + 0j) * (Z(3) @ Z(7))
+            + (0.355 + 0j) * (Z(4) @ Z(6))
+            + (0.355 + 0j) * (Z(4) @ Z(7))
+            + (0.355 + 0j) * (Z(5) @ Z(6))
+            + (0.355 + 0j) * (Z(5) @ Z(7)),
+        ),
+        (
+            "Square",
+            [3, 3],
+            -1.23,
+            2.34,
+            1.42,
+            True,
+            (0.615 + 0j) * (Y(0) @ Z(1) @ Y(2))
+            + (0.615 + 0j) * (X(0) @ Z(1) @ X(2))
+            + (0.615 + 0j) * (Y(1) @ Z(2) @ Y(3))
+            + (0.615 + 0j) * (X(1) @ Z(2) @ X(3))
+            + (0.615 + 0j) * (Y(0) @ Z(1) @ Z(2) @ Z(3) @ Y(4))
+            + (0.615 + 0j) * (X(0) @ Z(1) @ Z(2) @ Z(3) @ X(4))
+            + (0.615 + 0j) * (Y(1) @ Z(2) @ Z(3) @ Z(4) @ Y(5))
+            + (0.615 + 0j) * (X(1) @ Z(2) @ Z(3) @ Z(4) @ X(5))
+            + (0.615 + 0j) * (Y(0) @ Z(1) @ Z(2) @ Z(3) @ Z(4) @ Z(5) @ Y(6))
+            + (0.615 + 0j) * (X(0) @ Z(1) @ Z(2) @ Z(3) @ Z(4) @ Z(5) @ X(6))
+            + (0.615 + 0j) * (Y(1) @ Z(2) @ Z(3) @ Z(4) @ Z(5) @ Z(6) @ Y(7))
+            + (0.615 + 0j) * (X(1) @ Z(2) @ Z(3) @ Z(4) @ Z(5) @ Z(6) @ X(7))
+            + (0.615 + 0j)
+            * (
+                Y(0)
```

Comment:
You can keep them as one line and prevent black from reformatting using [fmt](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#ignoring-sections).